### PR TITLE
[flink] Support watermark change by altering table

### LINKED ISSUE #1075: [Feature] AlterTable support change watermark on flink 1.17
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Flink 1.17 support change watermark by altering table, which is not implemented by Paimon.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by schnappi17**
### Purpose

fix #1075 

### Tests

SchemaChangeITCase#testAddWatermark
SchemaChangeITCase#testDropWatermark
SchemaChangeITCase#testModifyWatermark

### CODE REVIEW

File: docs/content/how-to/altering-tables.md [hunk 1]
```diff
@@ -409,3 +409,51 @@ ALTER TABLE my_table ALTER COLUMN col_a SET DATA TYPE DOUBLE;
...
+## Adding watermark
+
+The following SQL adds a watermark with strategy `ts - INTERVAL '1' HOUR` on column `ts` which is marked as event time attribute of table `my_table`.
+
+{{< tabs "add-watermark" >}}
+
+{{< tab "Flink" >}}
+
+```sql
+ALTER TABLE my_table ADD WATERMARK FOR ts AS ts - INTERVAL '1' HOUR
```
Comment (JingsongLi) [inline docs/content/how-to/altering-tables.md:422]: Can you also add an example: `add computed column` here? Sometimes watermark is closely related to computed column.

**Review by JingsongLi — COMMENTED** [review]
Thanks @schnappi17 for the contribution, it looks very nice, just left one comment.

**Comment by schnappi17** [comment]
> Thanks @schnappi17 for the contribution, it looks very nice, just left one comment.

Thanks @JingsongLi for your suggestion, it looks nice for users. Updated, please help to take a look.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**